eth/tracers/native/call_flat.go

```diff
@@ -0,0 +1,297 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package native
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/tracers"
+)
+
+//go:generate go run github.com/fjl/gencodec -type flatCallFrame -field-override flatCallFrameMarshaling -out gen_flatcallframe_json.go
+//go:generate go run github.com/fjl/gencodec -type flatCallAction -field-override flatCallActionMarshaling -out gen_flatcallaction_json.go
+//go:generate go run github.com/fjl/gencodec -type flatCallResult -field-override flatCallResultMarshaling -out gen_flatcallresult_json.go
+
+func init() {
+	tracers.DefaultDirectory.Register("flatCallTracer", newFlatCallTracer, false)
+}
+
+// flatCallFrame is a standalone callframe.
+type flatCallFrame struct {
+	Action              flatCallAction  `json:"action"`
+	BlockHash           *common.Hash    `json:"blockHash"`
+	BlockNumber         uint64          `json:"blockNumber"`
+	Error               string          `json:"error,omitempty"`
+	Result              *flatCallResult `json:"result,omitempty"`
+	Subtraces           int             `json:"subtraces"`
+	TraceAddress        []int           `json:"traceAddress"`
+	TransactionHash     *common.Hash    `json:"transactionHash"`
+	TransactionPosition uint64          `json:"transactionPosition"`
+	Type                string          `json:"type"`
+}
+
+type flatCallFrameMarshaling struct {
+	BlockNumber         hexutil.Uint64
+	TransactionPosition hexutil.Uint64
+}
+
+type flatCallAction struct {
+	Author         *common.Address `json:"author,omitempty"`
```

Comment:
These fields are not used in the code, we should remove them:

- Author
- RewardType

Also `CreationMethod` but we could fill that in (I suppose it should distinguish between txcreate, create, or create2).